### ISSUE TITLE
Allow ROW Permit Applicants to Pay DS fees online for VOID Status

### ISSUE THREAD

**Issue by TracyLinder** [cityofaustin/atd-data-tech#4071]
### As a permit applicant, I need to pay DS permit fees online for permits with a status Void.

**Stakeholders:** Paloma, Kim

**Acceptance Criteria:**
- Public Portal users should be able to view and make payments on DS folders with status  'Void'

**Notes:**
- We tested this online; DS payments cannot be made when folder status is VOID. (5/3/21)
- Surya mentioned the ability to view folders in portal is controlled based off lookup table 99052. She believed adding two new rows here for these statuses should fix the issue. 
- Can the functionality be modified to only view Withdrawn or Void folders ONLY if they have outstanding fees? Need estimate from developers. 
    - If this is possible, I'd (Tracy) prefer it be developed in a separate issue. 

Robert Notes:
Insert records into 99052:
Lookupstring = 'RW'
Lookup2 (StatusCode) = 50045
LookupString2 (StatusDesc) = Withdrawn
LookupString3  = 'Y'

Lookupstring = 'RW'
Lookup2 (StatusCode) = 56050
LookupString2 (StatusDesc) = VOID
LookupString3  = 'Y'

Lookupstring = 'DS'
Lookup2 (StatusCode) = 50045
LookupString2 (StatusDesc) = Withdrawn
LookupString3  = 'Y'

Lookupstring = 'DS'
Lookup2 (StatusCode) = 56050
LookupString2 (StatusDesc) = VOID
LookupString3  = 'Y'

Lookupstring = 'EX'
Lookup2 (StatusCode) = 50045
LookupString2 (StatusDesc) = Withdrawn
LookupString3  = 'Y'

Lookupstring = 'EX'
Lookup2 (StatusCode) = 56050
LookupString2 (StatusDesc) = VOID
LookupString3  = 'Y'

**Comment by vschandramouli** [comment]
Moving ticket to Backlog.

**Comment by licarir** [comment]
![image.png](https://images.zenhubusercontent.com/5ebae52ece896d5e6d9acd69/a59ce859-a705-4f5c-81ca-d523c9f52523)

**Comment by licarir** [comment]
@TracyLinder @Nadin-Nader This has been requested to be moved to Test. I will be looking at the other items on my list to see what I can get to PROD before the 20th.

**Comment by TracyLinder** [comment]
@licarir what is the deployment task #? Will you please update this when it's ready for UAT?

**Comment by licarir** [comment]
@TracyLinder There is not a ServiceNow ticket. I put the card directly on their board. I will check it daily and let you all know when it is ready for UAT.

**Comment by TracyLinder** [comment]
Great, thanks Robert.

Tracy Linder (she/her)
Product Manager, Data & Technology Services
Austin Transportation Department
——————————————————————
data.mobility.austin.gov<https://gcc01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdata.mobility.austin.gov%2F&data=02%7C01%7CTracy.Linder%40austintexas.gov%7C7945cc805599404f4caf08d8359631ff%7C5c5e19f6a6ab4b45b1d0be4608a9a67f%7C0%7C0%7C637318266481081651&sdata=2eb6bM92Yns5NEu7hvUCknL3Ao%2BE95ja7YZB2NQtTao%3D&reserved=0>

**Comment by licarir** [comment]
@TracyLinder @Nadin-Nader This has made it to Test and is ready for UAT.

**Comment by TracyLinder** [comment]
### UAT Results (FAILED)

- ❌ RSN 12301964 (EX)
    - We tried both Void & Withdrawn status and I was unable to view the folder on portal (My Bills or My permits)

**Comment by licarir** [comment]
@TracyLinder Hello Tracy, I believe there is something off about the folder you picked. I worked with Swapna on the Portal team on Test and we got three folders (RW, DS, and EX) to show up on Portal, all with Void and Withdrawn statuses. Please try another folder when you find time. Thank you.

**Comment by TracyLinder** [comment]
This has been assigned to LIT. @aishwarya2505 & @SubhaShroff will do some research.

**Comment by SubhaShroff** [comment]
- I tested for this ticket yesterday(4/7/2021) on TEST instance.
 Currently
 Any Folders with WITHDRAWN, VOID status can't be seen on 'My Permits/Cases' as the page is designed to allow only limited folder statuses to display on this portal page. [Need to check what are the folder statuses that are allowed to display.]

If the bill is generated, for EX, DS, RW with withdrawn or VOID status, the folder can be seen under 'My Bills'  and allowed to pay the fee online.
Once the fee is paid it won't be available even under 'My Bills'. (I tested for RSN 12301964 (EX) also. It has VOID status and there is unpaid bill. So now the logged in user can see it on portal under MY Bills.
-------------------------------------------------------------------------------------
If the user still wants to see those folders under my permits/cases, after payment:
 Change the folder status with online payment:
For example:
WITHDRAWN => Withdrawn Paid
VOID => VOID Paid 
That might work.

**Comment by TracyLinder** [comment]
**Stakeholder Review & Update**
- Based on Subha's research, customers are already able to pay bills for void or withdrawn folder statuses. If this is true, then we can close this ticket. Paloma & Tracy will test to be sure.

**Comment by TracyLinder** [comment]
Paloma & I will test this together on Monday 5/3 @ 1pm.

**Comment by TracyLinder** [comment]
### Update

I tested this with Paloma & Kim today. The portal doesn't allow users to pay fees on a VOID DS folder. I updated the issue name & description to focus only on DS-VOID payments, and putting this back on deck.

**Comment by SubhaShroff** [comment]
### Status: I need to look at VOID DS folder payment as per Tracy's update.

**Comment by SubhaShroff** [comment]
### Status:
Today started to work to fix it.

**Comment by SubhaShroff** [comment]
@TracyLinder @Nadin-Nader Did the necessary change on local branch. It can be tested on TEST portal as it is payment related. So, first moving it to DEV and see if it can be seen atleast under 'My Bills'.
Submitted DEV promotion request : **SCTASK0137291** (f057bd98d1c3).

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @TracyLinder  The DEV deployment is done and I tested it on DEV portal. Though we can't make payments on DEV portal, we can atleast see that DS VOID (11459070) permit is displayed under 'My Bills'.

![image](https://user-images.githubusercontent.com/51680715/118154402-9e016280-b3dc-11eb-8e87-4e9730478893.png)

**Comment by Nadin-Nader** [comment]
@SubhaShroff Per Paloma, we can move this to TEST to test the payment functions

**Comment by SubhaShroff** [comment]
@Nadin-Nader  TEST Deployment request is submitted - **SCTASK0137746** (for commit-f057bd98d1c3)

**Comment by SubhaShroff** [comment]
@Nadin-Nader  tested on TEST. Now DS Void can be paid from 'My Bills'.
![image](https://user-images.githubusercontent.com/51680715/118550847-d5487a00-b722-11eb-8d07-59ae44e61cde.png)
![image](https://user-images.githubusercontent.com/51680715/118551055-150f6180-b723-11eb-87ed-fb660677b29c.png)

**Comment by SubhaShroff** [comment]
### Status:
Waiting for TEST UAT.

**Comment by SubhaShroff** [comment]
### Status:
Waiting for TEST UAT.

**Comment by Nadin-Nader** [comment]
UAT in weekly meeting with StakeHolders

**Comment by Nadin-Nader** [comment]
Need to ReUAT  due to webpage loading issues

**Comment by Nadin-Nader** [comment]
##  **Status:**

TEST UAT Passed. 

## **APPROVED for PROD Deployment**

**Comment by SubhaShroff** [comment]
@Nadin-Nader  **SCTASK0140829** is submitted for PROD Deployment.

**Comment by Nadin-Nader** [comment]
## **Status:**

Still waiting on CTM DevOps to deploy to PROD.

**Comment by SubhaShroff** [comment]
### Status:
PROD deployment is done last evening. 🎉🎉

**Comment by Nadin-Nader** [comment]
Sent email to Stakeholders informing them that this is on PROD and to keep an eye on it for any issues (see email below). I will keep this open til Monday 6/14/21. I will follow up then to see if there are issues or if i can close the card.

> From: Nader, Nadin 
> Sent: Tuesday, June 8, 2021 12:03 PM
> To: Amayo-Ryan, Paloma <Paloma.Amayo-Ryan@austintexas.gov>; Perez, Kimberly <Kimberly.Perez@austintexas.gov>
> Cc: Pena, Isabell <Isabell.Pena@austintexas.gov>; Linder, Tracy <Tracy.Linder@austintexas.gov>
> Subject: Allow ROW Permit Applicants to Pay DS fees online for VOID Status #4071 on PROD
> 
> Hello Everyone,
> 
> I hope you are all doing well. I wanted to let you know that the request to Allow ROW Permit Applicants to Pay DS fees online for VOID Status, Card #4071 is now on PROD for the applicant to use and for yall to verify that it is working well on PROD. 
> 
> I will keep this open for now and check back on Monday to see if there are any issues. If there aren’t, I will go ahead and close the card
> 
> Respectfully,
> 
> Nadin A. Nader (He/Him/His)
> IT Business Systems Analyst
> Data and Technology Services
> ————————————————————————————
> transportation.data@austintexas.gov 🚦 austinmobility.io

**Comment by SubhaShroff** [comment]
### Status:
On PROD. Waiting for stakeholders' response.

**Comment by SubhaShroff** [comment]
Status:
On PROD. Waiting for stakeholders' response.

**Comment by Nadin-Nader** [comment]
Have not heard about any issues with this. Sending email and closing:

> From: Nader, Nadin 
> Sent: Thursday, June 17, 2021 9:05 AM
> To: Amayo-Ryan, Paloma <Paloma.Amayo-Ryan@austintexas.gov>; Perez, Kimberly <Kimberly.Perez@austintexas.gov>
> Cc: Pena, Isabell <Isabell.Pena@austintexas.gov>
> Subject: RE: Allow ROW Permit Applicants to Pay DS fees online for VOID Status #4071 on PROD
> 
> Good Morning,
> 
> Hope you are all doing well. I was reaching out to let you know that I will be closing this out as I have not heard that we are having trouble with Applicants Paying DS fees online for VOID Statuses. If there are any issues concerning this, please feel free to email me and let me know.
> 
> From: Amayo-Ryan, Paloma <Paloma.Amayo-Ryan@austintexas.gov> 
> Sent: Tuesday, June 8, 2021 12:25 PM
> To: Nader, Nadin <Nadin.Nader@austintexas.gov>; Perez, Kimberly <Kimberly.Perez@austintexas.gov>
> Cc: Pena, Isabell <Isabell.Pena@austintexas.gov>; Linder, Tracy <Tracy.Linder@austintexas.gov>
> Subject: RE: Allow ROW Permit Applicants to Pay DS fees online for VOID Status #4071 on PROD
> 
> Thanks Nate!  We’ll keep an eye out for issues.
> 
> Paloma Amayo-Ryan
> Program Manager
> (512) 974-2841
> 
>  	Austin Transportation Department 
> Right of Way Management Division
> Mail: P.O. Box 1088, Austin, TX 78767
> Join our Mailing List
> Call 3-1-1 to report your right of way closure
> Read our Mobility Guidelines and Permit Conditions
> http://twitter.com/AustinMobility
> http://facebook.com/ATXTransportation
> 
> 
> 
> From: Nader, Nadin 
> Sent: Tuesday, June 8, 2021 12:03 PM
> To: Amayo-Ryan, Paloma <Paloma.Amayo-Ryan@austintexas.gov>; Perez, Kimberly <Kimberly.Perez@austintexas.gov>
> Cc: Pena, Isabell <Isabell.Pena@austintexas.gov>; Linder, Tracy <Tracy.Linder@austintexas.gov>
> Subject: Allow ROW Permit Applicants to Pay DS fees online for VOID Status #4071 on PROD
> 
> Hello Everyone,
> 
> I hope you are all doing well. I wanted to let you know that the request to Allow ROW Permit Applicants to Pay DS fees online for VOID Status, Card #4071 is now on PROD for the applicant to use and for yall to verify that it is working well on PROD. 
> 
> I will keep this open for now and check back on Monday to see if there are any issues. If there aren’t, I will go ahead and close the card
> 
> Respectfully,
> 
> Nadin A. Nader (He/Him/His)
> IT Business Systems Analyst
> Data and Technology Services
>
